### DISCUSSION TITLE
Email noise reduction for failing tasks

### DISCUSSION THREAD

**Discussion by metra** [apache/airflow#29483] [Ideas]
**Description**

Failing tasks can create a lot of email noise if you have both `email_on_failure` and `sla` enabled.

**Use case / motivation**

Outside of working hours, say 6pm through 10am, I can receive 16*2 email notifications. Every hour, one email for the failure and one email for the missed SLA.

Two ideas:

- Sending both the SLA miss and the failure email feels redundant.  If a task is failing then it should be obvious that it's also missing its SLA. Perhaps skip the SLA email in this case?
- The SLA miss emails are threaded because their subject lines are equivalent. However, the failure emails include the failure date and time, which breaks threading. Consider dropping the time and/or date from the failure email?

**Related Issues**

None that I could find.

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Sounds a nice feature @metra . Maybe you would like to contribute it yourself ? We are happy to guide you through contribution process.
